### PR TITLE
LPS-182261 [TR2] Automation Test | master

### PR DESCRIPTION
**Pull request by joelrocha68**
Product Version Permissions
https://issues.liferay.com/browse/LPS-182259

### CODE REVIEW

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			RightClick.rightClick(
+				key_projectName = ${projectName},
+				locator1 = "Testray#ITEM_NAME");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:80]: Please, add the first call of this block in the `Given`. Only the action needs to stay in this block.
Exemple: 
```
task ("When the user go to the product version page") {
			RightClick.rightClick(
				key_projectName = ${projectName},
				locator1 = "Testray#ITEM_NAME");

			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
		}
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+			RightClick.rightClick(
+				key_projectName = ${projectName},
+				locator1 = "Testray#ITEM_NAME");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+
+			RightClick.rightClick(
+				key_projectName = ${productVersion},
+				locator1 = "Testray#ITEM_NAME");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:114]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:150]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+		task ("When the user create the product version") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			TestrayManage.createProductVersion(
+				manageOption = "Manage Product Versions",
+				nameOptions = ${projectName},
+				version = "7.4");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:181]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+				valuePagination = 150);
+
+			TestrayContextMenu.clickContextMenuProjects(name = ${projectName});
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+
+			TestrayManage.deleteProductVersion(
+				itemManage = "Delete",
+				versionName = ${productVersion});
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:215]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+				valuePagination = 150);
+
+			TestrayContextMenu.clickContextMenuProjects(name = ${projectName});
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+
+			TestrayManage.editProductVersion(
+				caseName = "3.4",
+				itemManage = "Edit",
+				valueName = ${productVersion});
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:249]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			RightClick.rightClick(
+				key_projectName = ${projectName},
+				locator1 = "Testray#ITEM_NAME");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:282]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+			RightClick.rightClick(
+				key_projectName = ${projectName},
+				locator1 = "Testray#ITEM_NAME");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Product Versions");
+
+			RightClick.rightClick(
+				key_projectName = ${productVersion},
+				locator1 = "Testray#ITEM_NAME");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:316]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayProductVersionPermissions.testcase:352]: Same comment as above.

**Review by luanakarla — CHANGES_REQUESTED** [review]
Hey @joelrocha68 , thanks for working on it!

I have given you some suggestions, can you please take a look?

**Review by luanakarla — CHANGES_REQUESTED** [review]
Hey @joelrocha68 , thanks for working on it!

I have given you some suggestions, can you please take a look?

**Review by luanakarla — APPROVED** [review]

**Comment by luanakarla** [comment]
LGTM!

**Comment by joelrocha68** [comment]
ci:forward

**Pull request closed without merging**